pkg/manager/manager_arp.go

```diff
@@ -0,0 +1,155 @@
+package manager
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/kamhlos/upnp"
+	"github.com/plunder-app/kube-vip/pkg/cluster"
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+)
+
+// Start will begin the Manager, which will start services and watch the configmap
+func (sm *Manager) startARP() error {
+	var cpCluster *cluster.Cluster
+	var ns string
+	var err error
+
+	// use a Go context so we can tell the leaderelection code when we
+	// want to step down
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// listen for interrupts or the Linux SIGTERM signal and cancel
+	// our context, which the leader election code will observe and
+	// step down
+	signalChan := make(chan os.Signal, 1)
+	// Add Notification for Userland interrupt
+	signal.Notify(signalChan, syscall.SIGINT)
+
+	// Add Notification for SIGTERM (sent from Kubernetes)
+	signal.Notify(signalChan, syscall.SIGTERM)
+
+	// Add Notification for SIGKILL (sent from Kubernetes)
+	signal.Notify(signalChan, syscall.SIGKILL)
+
+	// Shutdown function that will wait on this signal, unless we call it ourselves
+	go func() {
+		<-signalChan
+		log.Info("Received termination, signaling shutdown")
+		if sm.config.EnableControlPane {
+			cpCluster.Stop()
+		}
+		// Cancel the context, which will in turn cancel the leadership
+		cancel()
+	}()
```

Comment:
It might be good to push the signal handling up to the `cmd` package and plumb through a cancelable context or stop channel into the manager bits.

Reply:
Agreed, This needs simplifying as it's causing a panic on graceful shutdown :-( 
The shutdown triggers a close on a channel within the HA and now the services goroutine. Might need some help here to simplify it.